addons/mail/models/mail_tracking_value.py

```diff
@@ -101,8 +101,8 @@ def _create_tracking_values(self, initial_value, new_value, col_name, col_info,
             })
         elif col_info['type'] in {'one2many', 'many2many'}:
             values.update({
-                'old_value_char': ', '.join(initial_value.mapped('display_name')) if initial_value else '',
-                'new_value_char': ', '.join(new_value.mapped('display_name')) if new_value else '',
+                'old_value_char': ', '.join(value.display_name or f"{value._name},{value.id}" for value in initial_value) if initial_value else '',
+                'new_value_char': ', '.join(value.display_name or f"{value._name},{value.id}" for value in new_value) if new_value else '',
```

Comment:
Single quotes for consistency? :disguised_face: 
```suggestion
                'old_value_char': ', '.join(value.display_name or f'{value._name},{value.id}' for value in initial_value) if initial_value else '',
                'new_value_char': ', '.join(value.display_name or f'{value._name},{value.id}' for value in new_value) if new_value else '',
```
```suggestion
                'old_value_char': ', '.join(value.display_name or f"{value._name},{value.id}" for value in initial_value) if initial_value else '',
                'new_value_char': ', '.join(value.display_name or f"{value._name},{value.id}" for value in new_value) if new_value else '',
```